bus/EasyCaching.Bus.CSRedis/EasyCaching.Bus.CSRedis.csproj

```diff
@@ -36,6 +36,7 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\src\EasyCaching.Core\EasyCaching.Core.csproj" />
+    <ProjectReference Include="..\..\src\EasyCaching.Decoration.Polly\EasyCaching.Decoration.Polly.csproj" />
```

Comment:
`EasyCaching.Decoration.Polly` should be added by users.

`EasyCaching.Core` should have the default/empty implementation of decoration. 

If user add `EasyCaching.Decoration.Polly`, they will use `Polly` to done CB or FB.

If not, they will use the default/empty implementation.

It might be better to give users a choice.

Reply:
Default implementation would require us to implement some of Polly functionality by ourselves. I don't think it's worth it.

Empty implementation of decoration will mislead people. They may configure `options.DecorateWithCircuitBreaker` without adding Polly and will think, that it works when it does nothing.

We can remove Polly from Buses, but then there won't be retry and fallbacks by default. Maybe it's really better to always configure retry and fallbacks explicitly. Fallback by default may also mislead people because they may expect a error from bus to throw exception instead of silently do nothing when Redis is down.
What do you think about this?

Reply:
This will force users to depend on the `Polly` library. Sometimes users don't need the corresponding function, but depending on it will cause their confusion.

In addition, as far as I know, many companies have their own internal class libraries that include CB and FB, but they are not open source. 

Therefore, the implementation of this feature still does not want to directly depend on `Polly` library in `EasyCaching.Core`.

Reply:
I'm suggesting to remove Polly reference and retry/fallback by default: https://github.com/dotnetcore/EasyCaching/pull/284/commits/c736d9ec6ef1d00d264d1c07c9ad2aadca3fd555

When user don't need resilience, they don't need to depend on Polly.

If they want retry, fallback or circuit breaker, they have two options:
1. Reference `EasyCaching.Decoration.Polly` and use `.DecorateWith` extensions from it
2. Make their own Decorator implementation and configure it like this:
```
options.Decorate((name, serviceProvider, cachingBusFactory) =>
    () => new CustomBusRetryDecorator(name, cachingBusFactory));
```

Reply:
Maybe the following solution is more better.

Add a new provider that support circuit breaker and fallback.

```cs
public class CircuitBreakerAndFallbackProvider : IEasyCachingProvider
{
//  custom implementation
}
```

And use this new provider to replace the default one.

```diff
            services.AddEasyCaching(option =>
            {
--               option.UseXxx("default");
++               option.UseXxxWithCircuitBreakerAndFallback("cus");
            });
```

Reply:
It will require us to write `UseXxxWithCircuitBreakerAndFallback` extension for each provider. 
If someone adds new decorator, he can't just use is for any provider.

Reply:
Although in case of hybrid cache we need CircuitBreaker without fallback on distributed cache. So combining decorators - is important feature.

Reply:
Could you describe your offer in more detail? Maybe I don't fully understand it.